packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx

```diff
@@ -196,7 +196,7 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
     // If the returned value is not an array then check to see if it's a promise or PromiseLike. If it is then resolve it asynchronously.
     if (Array.isArray(suggestionsArray)) {
       this.resolveNewValue(updatedValue, suggestionsArray);
-    } else if (suggestionsPromiseLike.then) {
+    } else if (suggestionsPromiseLike) {
```

Comment:
I might say that wrapping it it all in ` if (suggestions) ` because really we should check to ensure that suggestions exists. IsArray seems to check that for now but it seems safer to wrap the entire thing in it. Thanks for fixing this!

I agree that pickers should be named differently, and really suggestions and autofill should be split out into their own top level components as well.